gflib/bg.c

```diff
@@ -259,7 +259,7 @@ static void SetBgAffineInternal(u8 bg, s32 srcCenterX, s32 srcCenterY, s16 dispC
             return;
         break;
     case 2:
-        if (bg < 2 || bg > 3)
+        if (bg < 2 || bg >= NUM_BACKGROUNDS)
```

Comment:
```suggestion
        if (bg != 2 && bg != 3)
```
This makes more sense because not only does this match, but it indicates that case 2 ONLY works on bg numbers 2 and 3.